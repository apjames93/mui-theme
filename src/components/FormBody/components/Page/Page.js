import React from 'react';
import MuiStoryblok from 'mui-storyblok';
import PropTypes from 'prop-types';

const Page = ({ form }) => {
  const formState = form.getState();
  return (
    <MuiStoryblok
      theme={formState.values.theme}
      accessToken={formState.values.storyBlokAccessToken}
      version="draft"
    />
  );
};

export default Page;

Page.propTypes = {
  form: PropTypes.shape({
    getState: PropTypes.func,
  }),
};

Page.defaultProps = {
  form: {},
};
