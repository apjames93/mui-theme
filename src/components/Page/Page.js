import React, { useState, useEffect } from 'react';
import { MuiStoryblok } from 'mui-storyblok';
import PropTypes from 'prop-types';

export const Page = ({ accessToken, theme, pageRedirect }) => {
  const [page, setPage] = useState(window.location.pathname);
  const [togglePage, setToggle] = useState(true);
  const toggleStoryblok = () => {
    setToggle(false);
    setTimeout(() => {
      setToggle(true);
    }, 100);
  };

  useEffect(() => {
    if (page !== pageRedirect) {
      const { host } = window.location;
      setPage(pageRedirect);
      window.history.pushState({}, null, `http://${host}/${pageRedirect}`);
      toggleStoryblok();
    }
  }, [accessToken, page, pageRedirect, togglePage]);

  return (
    <>
      {
        togglePage
        && (
        <MuiStoryblok
          useObjectTheme={true}
          theme={theme}
          accessToken={accessToken}
          version="draft"
        />
        )
      }
    </>
  );
};

export default Page;

Page.propTypes = {
  accessToken: PropTypes.string.isRequired,
  theme: PropTypes.shape().isRequired,
  pageRedirect: PropTypes.string.isRequired,
};

Page.defaultProps = {};
