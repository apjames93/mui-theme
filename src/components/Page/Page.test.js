import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { Page } from './Page';

function setup() {
  const props = {
    form: {
      getState: jest.fn(() => ({
        values: {
          theme: {},
          storyBlokAccessToken: 'wasd',
        },
      })),
    },
    theme: {},
    accessToken: '',
    pageRedirect: '/',
  };

  const comp = mount(<Page {...props} />);
  return { comp, props };
}

describe('<Page />', () => {
  it('renders Page', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create((<Page {...props} />));
    expect(tree).toMatchSnapshot();
  });
});
