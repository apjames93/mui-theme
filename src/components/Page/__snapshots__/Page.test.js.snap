// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    className="MuiCircularProgress-root makeStyles-root-4 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.5}
        strokeWidth={3}
        style={Object {}}
      />
    </svg>
  </div>
  
</div>
`;
